### DISCUSSION TITLE
Run Areas error

### DISCUSSION THREAD

**Discussion by krb0073** [GrossfieldLab/loos#75] [Q&A] [answered]
Hey All, 
When trying the run_areas.py code in the voronoi code I kept getting a error message . I was wondering if I was doing something wrong. 

My command
run_areas.py final.psf final.dcd 1000 0 10  30 'name == "P"' 'resname == "POPC"' 

the message that I get is:
Traceback (most recent call last):
  File "/group/jbmertz/software/loos/loos/Packages/Voronoi/run_areas.py", line 115, in <module>
    v.generate_voronoi()
  File "/gpfs20/group/jbmertz/software/loos/loos/Packages/Voronoi/Voronoi.py", line 146, in generate_voronoi
    self.voronoi = Voronoi(points)
  File "qhull.pyx", line 2601, in scipy.spatial.qhull.Voronoi.__init__
IndexError: tuple index out of range

Thanks for your time,
Kyle

**Comment by liyp1215** [comment]
Hi Kyle, did you solve this problem? I also encountered the same error

**Reply by agrossfield** [reply]
The first selection has to be a superset of all of the subsequent ones.

**Reply by liyp1215** [reply]
I jus solved this error - actually the selected atom I used is missing in my trajectory. Thanks!

**Comment by agrossfield** [comment, marked as answer]
I’m glad. When I have time, I need to go back and make the error messages (and documentation in general) better for the Voronoi stuff.

**Reply by liyp1215** [reply]
Hi Alan,

This sounds good! It would be very helpful on how to use it. By the way, I got much lower values of area per lipid than expected. The system I analyzed is one with protein embedded in a membrane with multiple types of lipids. I used the following command:

_python run_areas.py TMM-S1-rep1.psf TMM-S1-rep1.dcd 0 -45 45 30 '(segname == "AM" && \!hydrogen) || name =~ "^(P|P1|P3)$"' 'resname == "MDSPE" && name == "P"' 'resname == "MTOCL1" && name =~ "^(P1|P3)$"' 'resname == "POPI" && name == "P"' 'segname =~ "^APM2\d+" && name == "P"' 'segname =~ "^A2PM2\d+" && name == "P"' 'segname =~ "^APM6\d+" && name == "P"' 'segname =~ "^A2PM6\d+" && name == "P"'_

Here, segname AM is protein and other selections are just the different types of lipids. The cell is put at x=y=0 already. Do you know what might be the possible reason or do you have any suggestion on this?

Best,
Yupeng

**Reply by agrossfield** [reply]
I have a guess, especially if the areas are roughly half of what you would expect: it's grabbing the lipids from both leaflets (guessing from the z range of -45 to 45) but then treating them as if they're all in the same plane (if I recall correctly, this particular program filters on z once and then literally ignores the z coordinate afterward). The solution is to run the code twice, once with z range [-45 0] and once [0 45]. 

I might go a step further, and restrict the z ranges further to (conservatively) match the range of the phosphates, so that the protein selection range matches the Ps. Otherwise, if the protein bows outward in the middle, you'll artificially reduce the effective area available to the lipids. 

Hope this helps,

Alan

**Reply by agrossfield** [reply]
For what it's worth, until I get around to extending the documentation, definitely run `command-name --fullhelp`. Each of the Voronoi tools has a pretty lengthy description. Also, asking questions here should generally get a quick answer (I think I answered Kyle offline, which was an error).

**Reply by liyp1215** [reply]
This makes sense to me and is very helpful Alan! I really appreciate your help, thanks!
